modules/http-client/src/main/java/com/devonfw/module/httpclient/common/impl/AbstractAsyncServiceHttpClient.java

```diff
@@ -117,4 +119,39 @@ protected Object handleUnsupportedBody(Object body) {
    */
   protected abstract HttpRequest createRequest(ServiceClientInvocation<S> invocation);
 
+  @Override
+  protected <R> CompletableFuture<R> doCall(ServiceClientInvocation<S> invocation) {
+
+    long startTime = System.nanoTime();
+    HttpRequest request = createRequest(invocation);
+    CompletableFuture<HttpResponse<String>> future = this.client.getHttpClient().sendAsync(request,
+        BodyHandlers.ofString());
+    return future.thenApplyAsync(response -> handleResponse(response, startTime, invocation));
+
+  }
+
+  @SuppressWarnings({ "rawtypes", "unchecked" })
+  private <R> R handleResponse(HttpResponse<?> response, long startTime, ServiceClientInvocation<S> invocation) {
+
+    Throwable error = null;
+    String service = invocation.getServiceDescription(response.uri().toString());
+    try {
+      int statusCode = response.statusCode();
+      if (statusCode >= 400) {
+        error = createError(response, invocation, service);
+        this.errorHandler.accept(error);
+      } else {
+
+        Object result = createResult(response, invocation);
+        return (R) result;
+      }
+    } catch (Throwable t) {
+      this.errorHandler.accept(t);
+      error = t;
+    } finally {
+      ServiceClientPerformanceLogger.log(startTime, service, response.statusCode(), error);
+    }
+    return null;
+  }
```

Comment:
This method is not new. Is that a bug of github that it is showing this as newly added in the diffs?
Here is the existing method before this PR:
https://github.com/devonfw/devon4j/blob/3268e8e563ec1da885ad07467e41aa8cbfd904c5/modules/http-client/src/main/java/com/devonfw/module/httpclient/common/impl/AbstractAsyncServiceHttpClient.java#L57

@ssarmokadam There is IMHO nothing wrong from your end, I am just confused by what github is showing me to review...